google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacet.java

```diff
@@ -52,4 +56,51 @@ public static AppEngineStandardFacet getAppEngineFacetByModule(@Nullable Module
     return FacetManager.getInstance(module).getFacetByType(ID);
   }
 
+  /**
+   * Returns {@code true} if this app is targeting the Flexible Environment instead of the Standard
+   * Environment.
+   */
+  public boolean isFlexEnvironment() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "flex".equalsIgnoreCase(appEngineStandardWebApp.getEnv().getStringValue());
+  }
+
+  /**
+   * Returns {@code true} if this app is targeting the Managed VM environment instead of the
+   * Standard Environment.
+   */
+  public boolean isManagedVmEnvironment() {
```

Comment:
I remember that I exposed separate methods for env:flex and vm:true previously because the name of the environment is displayed differently on the web console and we wanted to reflect this in our deployment dialog too.